server/itx/server.go

```diff
@@ -221,6 +217,11 @@ func StartServer(ctx context.Context, svr *Server, probeSvr *probe.Server, cfg c
 	if err := adminserv.Shutdown(ctx); err != nil {
 		log.L().Error("Error when serving metrics data.", zap.Error(err))
 	}
+	if cfg.System.HeartbeatInterval > 0 {
+		if err := task.Stop(ctx); err != nil {
+			log.L().Panic("Failed to stop heartbeat routine.", zap.Error(err))
+		}
+	}
```

Comment:
Stop() should be called in the LIFO order of Start()

task starts after svr (and used svr), so should stop before svr